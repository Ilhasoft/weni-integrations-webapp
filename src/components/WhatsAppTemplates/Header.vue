<template>
  <div class="whatsapp-templates-header">
    <div class="whatsapp-templates-header__wrapper">
      <div class="whatsapp-templates-header__icon">
        <img :src="icon" />
      </div>
      <span class="whatsapp-templates-header__title">{{ title }}</span>
    </div>
    <unnnic-button size="small" type="secondary">Novo template</unnnic-button>
  </div>
</template>

<script>
  export default {
    name: 'WhatsAppTemplatesHeader',
    props: {
      title: {
        type: String,
        default: 'WhatsApp',
      },
      icon: {
        type: String,
        default: 'https://weni-sp-integrations-production.s3.amazonaws.com/wpp_oficial_YaNXiFt.svg',
      },
    },
  };
</script>

<style lang="scss" scoped>
  .whatsapp-templates-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: $unnnic-spacing-stack-md auto;

    &__wrapper {
      display: flex;
      align-items: center;
    }

    &__icon {
      display: flex;
      background-color: rgba(72, 172, 76, 0.2);
      border-radius: $unnnic-border-radius-sm;
      justify-content: center;
      margin-right: $unnnic-spacing-inline-sm;

      img {
        height: $unnnic-icon-size-md;
        width: $unnnic-icon-size-md;
        padding: $unnnic-inset-nano;
      }
    }

    &__title {
      font-family: $unnnic-font-family-primary;
      font-weight: $unnnic-font-weight-regular;
      font-size: $unnnic-font-size-title-md;
      line-height: $unnnic-line-height-md + $unnnic-font-size-title-md;
      color: $unnnic-color-neutral-darkest;
    }
  }
</style>
