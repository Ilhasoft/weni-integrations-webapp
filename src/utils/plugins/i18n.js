import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
