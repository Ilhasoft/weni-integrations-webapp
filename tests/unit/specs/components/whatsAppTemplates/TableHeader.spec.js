import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Header from '@/components/whatsAppTemplates/TableHeader.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('components/whatsAppTemplates/TableHeader.vue', () => {
  let wrapper;
  let store;
  let actions;
  let state;

  beforeEach(() => {
    actions = {
      getAppType: jest.fn(() => {
        return { data: { icon: 'icon' } };
      }),
      fetchWppProfile: jest.fn(() => {
        return { data: { photo_url: 'photo' } };
      }),
    };

    state = {
      appType: {
        currentAppType: { icon: 'icon' },
        loadingCurrentAppType: false,
      },
    };

    store = new Vuex.Store({
      actions,
      state,
    });

    wrapper = shallowMount(Header, {
      localVue,
      store,
      i18n,
      stubs: {
        UnnnicButton: true,
      },
      mocks: {
        $route: {
          params: {
            appCode: 'wpp-cloud',
            appUuid: '123',
          },
        },
        $router: {
          push: jest.fn(),
        },
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('navigateToCreateTemplate()', () => {
    it('should change route to create template', () => {
      const spy = spyOn(wrapper.vm.$router, 'push');
      expect(spy).not.toHaveBeenCalled();
      wrapper.vm.navigateToCreateTemplate();

      expect(spy).toBeCalledTimes(1);
      expect(spy).toHaveBeenCalledWith({
        path: `/apps/my/${wrapper.vm.$route.params.appCode}/${wrapper.vm.$route.params.appUuid}/templates/create`,
      });
    });
  });
});
