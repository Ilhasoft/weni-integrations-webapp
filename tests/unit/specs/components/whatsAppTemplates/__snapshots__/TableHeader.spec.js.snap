// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/whatsAppTemplates/TableHeader.vue should be rendered properly 1`] = `
<div class="whatsapp-templates-header">
  <div class="whatsapp-templates-header__wrapper"><span class="whatsapp-templates-header__title">
      WhatsApp Templates
    </span> <span class="whatsapp-templates-header__description">
      The following table shows the message templates currently configured for this number.
    </span></div>
  <unnnicbutton-stub type="secondary" class="whatsapp-templates-header__button">
    New Template
  </unnnicbutton-stub>
</div>
`;
